http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeoKubernetesServiceInterp.scala

```diff
@@ -0,0 +1,412 @@
+                                                                          userInfo,
+                                                                          AttachPersistentDisk,
+                                                                          googleProject)
+          _ <- if (hasAttachPermission) F.unit else F.raiseError[Unit](AuthorizationError(Some(userInfo.userEmail)))
+        } yield disk
+      case (createReq @ DiskConfigRequest.Create(name, _, _, _, _), false) =>
+        for {
+          ctx <- as.ask
+          diskOpt <- persistentDiskQuery.getActiveByName(googleProject, name).transaction
+          _ <- diskOpt.fold(F.unit)(d =>
+            F.raiseError[Unit](PersistentDiskAlreadyExistsException(googleProject, name, d.status, ctx.traceId))
+          )
+          hasPermission <- authProvider.hasProjectPermission(userInfo, CreatePersistentDisk, googleProject)
+          _ <- if (hasPermission) F.unit else F.raiseError[Unit](AuthorizationError(Some(userInfo.userEmail)))
+          samResource <- F.delay(PersistentDiskSamResource(UUID.randomUUID().toString))
+          runtimeServiceAccountOpt <- serviceAccountProvider
+            .getClusterServiceAccount(userInfo, googleProject)
+          _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done Sam call for getClusterServiceAccount")))
+          petSA <- F.fromEither(
+            runtimeServiceAccountOpt.toRight(new Exception(s"user ${userInfo.userEmail.value} doesn't have a PET SA"))
+          )
+          ctx <- as.ask
+          disk <- F.fromEither(
+            DiskServiceInterp.convertToDisk(userInfo,
+                                            petSA,
+                                            googleProject,
+                                            name,
+                                            samResource,
+                                            diskConfig,
+                                            CreateDiskRequest.fromDiskConfigRequest(createReq),
+                                            ctx.now)
+          )
+          _ <- authProvider
+            .notifyResourceCreated(samResource, userInfo.userEmail, googleProject)
+            .handleErrorWith { t =>
+              log.error(t)(
+                s"[${ctx.traceId}] Failed to notify the AuthProvider for creation of persistent disk ${disk.projectNameString}"
+              ) >> F.raiseError(t)
+            }
+          savedDisk <- persistentDiskQuery.save(disk).transaction
+
+        } yield savedDisk
+      case (_, true) =>
+        as.ask.flatMap(ctx => F.raiseError(DiskNotSupportedException(ctx.traceId)))
+    }
+
+  case class LeoKubernetesConfig(serviceAccountConfig: ServiceAccountProviderConfig,
+                                 clusterConfig: KubernetesClusterConfig,
+                                 nodepoolConfig: NodepoolConfig,
+                                 galaxyAppConfig: GalaxyAppConfig,
```

Comment:
I'm imagining that as these are extended, we will have a map of [AppType,A], but I think we should avoid designing the configs for this until we know how close/extendable our current config code is. In other words, I want to punt on how exactly we handle multiple app configs since I didn't want to be too predictive in terms of what they need. Similarly for nodepool configs, I think that we may want to have a map when we have more than one, and be really specific to galaxy here